tests/cases/compiler/transformsElideNullUndefinedType.ts

```diff
@@ -0,0 +1,32 @@
+// @target: es6
+
+var v0: null;
+var v1: undefined;
+
+function f0(): null { return null; }
+function f1(): undefined { return undefined; }
+
+var f2 = function (): null { return null; }
+var f3 = function (): undefined { return undefined; }
+
+var f4 = (): null => null;
+var f5 = (): undefined => undefined;
+
+function f6(p0: null) { }
+function f7(p1: undefined) { }
+
+class C<T> {
+    m0(): null { return null; }
+    m1(): undefined { return undefined; }

```

Comment:
can you also add tests for null and undefined in type position for parameters for methods, arrow functions, function expressions, and setters.